src/pages/es/guides/client-side-scripts.md

```diff
@@ -0,0 +1,195 @@
+
+2. Aunque un `<script>` solo se ejecuta una vez, el navegador ejecutará método `constructor()` de nuestro elemento cada vez que encuentre `<astro-heart>` en la página. Esto significa que puedes escribir seguramente código para un componente a la vez, incluso si tienes la intención de utilizar este componente varias veces en una página.
+
+📚 Puedes aprender más sobre elementos personalizaods en [la guía de Componentes Web Reusables de web.dev](https://web.dev/custom-elements-v1/) y en [la introducción de MDN a elementos personalizados](https://developer.mozilla.org/es/docs/Web/Web_Components/Using_custom_elements).
+
+### Pasando variables frontmatter a scripts
+
+En componentes Astro, el codigo en [el frontmatter](/es/core-concepts/astro-components/#script-de-un-componente) entre los cercos `---` se ejecuta en el servidor y no esta disponible en el navegador. Para mandar variables desde el servidor al cliente, necesitamos una manera de guardar nuestras variables y leerlas cuando JavaScript se ejecute en el navegador.
+
+Una manera para hacer esto es usando [atributos `data-*`](https://developer.mozilla.org/es/docs/Learn/HTML/Howto/Use_data_attributes) para guardar el valor de las vairables en tu HTML. Los scripts, incluyendo elementos personalizados, pueden entonces leer estos atributos usando la propiedad de elemento `dataset` una vez que tu HTML cargue en el navegador.
+
+En este componente ejemplo, un propiedad `message` es guardad en un atributo `data-message`, para que el elemento personalizado pueda leer `this.dataset.message` y obtener el valor de la propiedad en el navegador.
+
+```astro title="src/components/AstroGreet.astro" {2} /data-message={.+}/ "this.dataset.message"
+---
+const { message = 'Welcome, world!' } = Astro.props;
+---
+
+<!-- Guarda la propiedad message como un atributo data. -->
+<astro-greet data-message={message}>
+  <button>¡Saluda!</button>
+</astro-greet>
+
+<script>
+  class AstroGreet extends HTMLElement {
+    constructor() {
+			super();
+
+      // Lee el mensaje del atributo data.
+      const message = this.dataset.message;
+      const button = this.querySelector('button');
+      button.addEventListener('click', () => {
+        alert(message);
+      });
+		}
+  }
+
+  customElements.define('astro-greet', AstroGreet);
+</script>
+```
+Ahora podemos usar nuestro componente múltiples veces y ser saludados con un mensaje diferente por cada uno.
+
+```astro title="src/pages/example.astro"
+---
+import AstroGreet from '../components/AstroGreet.astro';
+---
+
+<!-- Usa el mensaje por defecto: “¡Bienvenido, mundo!” -->
+<AstroGreet />
+
+<!-- Usa mensajes personalizados pasados como propiedades. -->
+<AstroGreet message="¡Es un lindo día para diseñar componentes!" />
+<AstroGreet message="¡Me alegro de que lo lograras! 👋" />
+```
```

Comment:
The last tip of this page is missing in the translation:

" :::tip[Did you know?]
This is actually what Astro does behind the scenes when you pass props to a component written using a UI framework like React! For components with a `client:*` directive, Astro creates an `<astro-island>` custom element with a `props` attribute that stores your server-side props in the HTML output.
::: "